lms/djangoapps/course_api/serializers.py

```diff
@@ -32,6 +32,38 @@ def get_uri(self, course_overview):
         return getattr(course_overview, self.uri_attribute)
 
 
+class _AbsolutMediaSerializer(_MediaSerializer):  # pylint: disable=abstract-method
+    """
+    Nested serializer to represent a media object and its absolute path.
+    """
+    requires_context = True
+
+    def __call__(self, serializer_field):
+        self.context = serializer_field.context
+        return super(self).__call__(serializer_field)
+
+    url = serializers.SerializerMethodField(source="*")
```

Comment:
Having two fields named `uri` and `url` is confusing. Also, DRF has [special treatment](https://www.django-rest-framework.org/api-guide/serializers/#serializer_url_field) for fields named `url`. Can we call this `uri_absolute` ?

Reply:
This is a bit trickier situation in terms of keeping conventions. 

Although I see your point, since URL represents a reference to a web resource located on the network and URI tells the resource identifier, I'm not sure that we should change these conventions in favor to comply with DRF's implementation. Also, DRF's URL filed is only accessible when the serializer is using `HyperlinkedModelSerializer` as its base. Other than that, `uri_absolute` wouldn't conflict with DRF's `url` field anyway.

As this serializer would possibly never use `HyperlinkedModelSerializer` based on its nature, what would you think about leaving `url` and `uri` here, since it wouldn't conflict with the ModelSerializer and we could keep the general naming conventions. 😊

Reply:
If you can show me anywhere else in edx-platform that uses `uri` for relative URL and `url` for absolute URL, we can keep that, but otherwise I think it's introducing a new confusing convention. (Well, the real problem here is that `uri` should never have been used for relative paths in the first place; without a scheme part and a `:` it's neither a URL nor URI by any definition). I guess I've changed my mind and now my preference would be to put the absolute URL in `uri` and put the relative path or other identifier in `path` or `id`.

BTW are you 100% sure we need the relative URL to be returned for this use case? By far the cleanest solution is to remove the requirement for returning the relative path.

Reply:
As we discussed, I'm renaming the field fro `url` to `uri_absolute`.